// @aws-sdk/client-lex-runtime-service/dist-cjs/LexRuntimeService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LexRuntimeService = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const DeleteSessionCommand_1 = require("./commands/DeleteSessionCommand");
const GetSessionCommand_1 = require("./commands/GetSessionCommand");
const PostContentCommand_1 = require("./commands/PostContentCommand");
const PostTextCommand_1 = require("./commands/PostTextCommand");
const PutSessionCommand_1 = require("./commands/PutSessionCommand");
const LexRuntimeServiceClient_1 = require("./LexRuntimeServiceClient");
const commands = {
    DeleteSessionCommand: DeleteSessionCommand_1.DeleteSessionCommand,
    GetSessionCommand: GetSessionCommand_1.GetSessionCommand,
    PostContentCommand: PostContentCommand_1.PostContentCommand,
    PostTextCommand: PostTextCommand_1.PostTextCommand,
    PutSessionCommand: PutSessionCommand_1.PutSessionCommand,
};
class LexRuntimeService extends LexRuntimeServiceClient_1.LexRuntimeServiceClient {
}
exports.LexRuntimeService = LexRuntimeService;
(0, smithy_client_1.createAggregatedClient)(commands, LexRuntimeService);
